src/deskolem.ml

```diff
@@ -0,0 +1,49 @@
+open Core
+open Terms
+open Timed
+
+(** [subst_inv t s x] replaces the term starting with the symbol [s] by a fresh
+    variable [x] in the term [t]. *)
+let rec subst_inv : term -> sym -> term Bindlib.var -> term =
```

Comment:
Put the term argument at the end to later avoid expressions like `fun x' -> subst_inv x' s x`. With `subst_inv : sym -> term Bindlib.var -> term -> term`, you can replace `fun x' -> subst_inv x' s x` by `subst_inv s x`.

Reply:
Moreover, as s and x are invariant throughout the call, you can simplify and write:
```
let (*no rec*) subst_inv : sym -> term Bindlib.var -> term -> term = fun s x ->
  let rec subst t =
    ...
  in subst

Reply:
Fixed in 72cd2da3ee58df266cfb895a and 80d4d43bebf0b3863102ad0.